internal/dag/builder_test.go

```diff
@@ -2151,6 +2151,86 @@ func TestDAGInsert(t *testing.T) {
 		},
 	}
 
+	proxy37 := &projcontour.HTTPProxy{
```

Comment:
Nit: For proxy38 and proxy39, consider leaving a comment explaining why they are invalid.